src/containers/createRoom.js

```diff
@@ -9,35 +9,91 @@ export class createRoom extends Component {
     username: "",
   };
 
-  handleChange = (event) => {
+  handleChange(event) {
     this.setState({ [event.target.name]: event.target.value });
   };
 
-  handleSubmit = (event) => {
-    event.preventDefault();
+  getRooms() {
     const reqObj = {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
         Accept: "application/json",
       },
-      body: JSON.stringify({ room: this.state }),
+      body: JSON.stringify({
+        room: this.state
+      }),
     };
```

Comment:
You'll want to extract conceptual steps in your code to make both (1) the code inside these steps and (2) the code where these steps are invoked to be more legible.

It also encapsulates the logic of the function so that it has a clearer responsibility and has access to only what it needs. This makes code easier to change in the future, easier to read, and less likely to have bugs.